File: python/grade_item_functions.py

def modify_grade_max(grade_item, new_max):
    '''
    Function to try and update grade max.
    Preconditions:
        grade_item (GradeItem) : GradeItem object to be updated.
        new_max (string) : New maximum grade for the grade_item.
    Postconditions:
        If successful, The GradeItem is updated through the API.
        Returns:
            Errors (list) : Any and all errors that may have occured.
    '''
     
    errors = []
    if new_max is None:
        errors.append({'msg':'Missing new grade item maximum'})
    else:
        try:
            new_max = float(new_max)
        except:
            errors.append({'msg':'Grade maximum must be a number'})
            return errors
                   
        if new_max <= grade_item.get_max():
            errors.append({'msg':'New grade maximum must be larger than current maximum'})
        
        else:
            try:
                grade_item.set_max( new_max )
                return new_max
            except Exception as e:
                errors.append({'msg':str(e)})
    return errors

File: php/includes/functions/grade_item_functions.php
<?php 

require_once('../../wrapper/obj/API.php');
require_once('../../wrapper/obj/User.php');
require_once('../../wrapper/obj/GradeItem.php');
/**
 * Error checking for max. If success, updates max grade value
 * @param {Integer} grade_item_id
 * @param {Integer} max
 * @return {Array} Array of errors and error messages to be sent to frontend
 */
function modify_grade_max($course_id, $grade_item_id, $max)
{
	$user = unserialize($_SESSION['user']);
	$course = $user->get_course($course_id);
	$grade_item = $course->get_grade_item($grade_item_id);
	

	// Error object to return
	$errors = array();
	if ($max === '')
	{
		$errors[] = array ( 
			'msg' => 'Missing grade maximum'
		);
	}
	else if (!is_numeric($max))
	{
		$errors[] = array ( 
			'msg' => 'Grade maximum must be a number'
		);
	}
	else if (floatval($max) <= $grade_item->get_max())
	{
		$errors[] = array ( 
			'msg' => 'New grade maximum must be larger than current maximum'
		);
	}

	if (sizeof($errors) == 0)
	{
		$grade_item->set_max((float)$max);
		//store user that contains modified gradeitem
		$_SESSION['user'] = serialize($user);
		return put_grade_item($grade_item);
	}
	return $errors;
}

?>
